### PR TITLE
Classifier server

### PR DESCRIPTION
**Pull request by pnowosie**
Part of [APPS-33](https://golemproject.atlassian.net/browse/APPS-33)

This PR is to ensure that `frontend` can call `api` in the weirdest cases (CORS etc) and to catch possible issues early.
Request of POST `/api/classify` should now carry **form data** with either of two possible fields ([see specs](https://github.com/golemfactory/yagna-service-poc/pull/3/commits/be2cdfcb832c63a019e7be16e869c2ba74ed3d00)):
- text_file - blob of text file
- text - string, e.g. textarrea value, when blob is not supported

⚠️ No data validation is done at the moment.

Dear reviewer should be able to run api container with 
```
cd server/api
make

# consider to clean up
make clean
```

## Tests
```
# Send simple string form's field
http --form POST http://0.0.0.0:8001/api/classify text="Ala ma kota, psa, chomika, papugę i anginę" 
HTTP/1.0 200 OK

{
    "results": [
        {
            "classifier_id": "distilbart-cnn-12-6",
            "summary_text": "ala ma kota, psa, chomika, papugę i anginę"
        },
        {
            "classifier_id": "bert-base-uncased-emotion",
            "summary_list": [  ...


# Submit a file
http --form POST http://0.0.0.0:8001/api/classify text_file=@text.txt   
HTTP/1.0 200 OK

{
    "results": [
        {
            "classifier_id": "distilbart-cnn-12-6",
            "summary_text": "the tower is 324 metres (1,063 ft) tall, about the"
        },
        {
            "classifier_id": "bert-base-uncased-emotion",
            "summary_list": [  ...


# CORS support (?)
http OPTIONS http://0.0.0.0:8001/api/classify Access-Control-Request-Method:POST Access-Control-Request-Headers:"origin, x-requested-with" Origin:aws.amazon.com
HTTP/1.0 200 OK
Access-Control-Allow-Headers: origin, x-requested-with
Access-Control-Allow-Methods: DELETE, GET, HEAD, OPTIONS, PATCH, POST, PUT
Access-Control-Allow-Origin: aws.amazon.com
Allow: POST, OPTIONS
Content-Length: 0
Content-Type: text/html; charset=utf-8
Date: Mon, 19 Apr 2021 19:46:57 GMT
Server: Werkzeug/1.0.1 Python/3.8.9
Vary: Origin
```

### CODE REVIEW

**Review by johny-b — APPROVED** [review]
Everything works as described (port is 8001 not 5000 as in CORS example)

File: server/api/src/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#
```
Comment (Wiezzel) [inline server/api/src/requirements.txt]: This file shouldn't be committed if it's auto-generated and removed by `make clean`.

Comment (pnowosie) [reply server/api/src/requirements.txt]: Oh wait, deleting requirements during clean is probably done by mistake.

The idea behind attaching the `requirements.txt` into codebase is that person with only docker installed is able to quickly run the app. WDYT? 

Drawback of this approach lies in pipenv which does not play nicely with docker... so `requirements.txt` is redundant way to patch docker path... On the other hand pipenv simplifies creation of the `requirements` file and can be scripted with make ... without it one needs to remember to activate the venv first ... I don't know much about python so maybe I go in circles here.

Comment (Wiezzel) [reply server/api/src/requirements.txt]: How about using a multi-stage Docker build and generating requirements in the builder?

Comment (pnowosie) [reply server/api/src/requirements.txt]: Do you mean builder image installs `pipenv` to generate `requirements.txt` and then uses `pip` to install deps ❓ 

Maybe I should revert to use just plain `pip` & `requirements.txt` to manage the dependencies, so no python is needed if you want to run the server in the container and we don't track the dependencies in multiple places 🤔

Comment (Wiezzel) [reply server/api/src/requirements.txt]: >Do you mean builder image installs pipenv to generate requirements.txt and then uses pip to install deps

That's what I meant.

But I'm :+1: for reverting to plain pip.

Comment (pnowosie) [reply server/api/src/requirements.txt]: Done in 3b86010ce1965ca64df97e8ae181c83f743d2a44

**Review by Wiezzel — APPROVED** [review]

**Review by albatrosica — APPROVED** [review]
Works as a charm :)

**Pull request merged**